x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/rare_view/metric_selection.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useContext, useEffect, useState } from 'react';
+import { EuiHorizontalRule, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+
+import { RareFieldSelector } from '../rare_field';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { RareJobCreator } from '../../../../../common/job_creator';
+import { RareDetector } from '../rare_detector';
+import { PopulationFieldSelector } from '../population_field';
+import { DetectorDescription } from './detector_description';
+import { RARE_DETECTOR_TYPE } from './rare_view';
+
+interface Props {
+  setIsValid: (na: boolean) => void;
+  setRareDetectorType(t: RARE_DETECTOR_TYPE): void;
+  rareDetectorType: RARE_DETECTOR_TYPE;
+}
+
+export const RareDetectors: FC<Props> = ({ setIsValid, rareDetectorType, setRareDetectorType }) => {
+  const { jobCreator: jc, jobCreatorUpdated } = useContext(JobCreatorContext);
+  const jobCreator = jc as RareJobCreator;
+  const [detectorValid, setDetectorValid] = useState(false);
+
+  useEffect(() => {
+    let valid = false;
+    if (jobCreator.rareField !== null) {
+      if (rareDetectorType === RARE_DETECTOR_TYPE.RARE) {
+        // Rare only requires a rare field to be set
+        valid = true;
+      } else if (jobCreator.populationField !== null) {
+        // all others need a need the population field to be set
+        valid = true;
+      }
+    }
+    setIsValid(valid);
+    setDetectorValid(valid);
+  }, [jobCreatorUpdated]);
+
+  return (
+    <>
+      <RareDetector onChange={setRareDetectorType} />
+      <>
+        <EuiHorizontalRule />
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <RareFieldSelector />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            {rareDetectorType !== RARE_DETECTOR_TYPE.RARE && <PopulationFieldSelector />}
```

Comment:
As in https://github.com/elastic/kibana/issues/90941, is there anything we can do to stop the user selecting the same field for `by`, `over` and `partition`?

![image](https://user-images.githubusercontent.com/7405507/122588282-cd983000-d056-11eb-964b-53351605ddc6.png)

Reply:
Fixed for rare wizard in https://github.com/elastic/kibana/pull/100390/commits/22622182f15ebdc46eaca908881f466df295ca53